Pointer arithmetic

I get a couple of compiler warnings about the pointer arithmetic in `spk.c`. For examples:
```
spk.c:260:24: warning: arithmetic on a pointer to void is a GNU extension [-Wpointer-arith]
        val = (void *)pl->map + sizeof(double) * (pl->one[m][n] - 1)
```
I've fixed this in 860c72c6c0d89c9206948d47fc6ee62e60fa54a4 but you've changed it back. I'm wondering why. Did that not work?

--- a/src/planets.c
+++ b/src/planets.c
@@ -312,7 +312,7 @@ int jpl_calc(struct _jpl_s *pl, struct mpos_s *now, double jde, double rel, int 
 
         // compute record number and 'offset' into record
         blk = (u_int32_t)((jde + rel - pl->beg) / pl->inc);
-        z = pl->map + (blk + 2) * pl->rec;
+        z = (double*)pl->map + (blk + 2) * pl->rec/sizeof(double);
         t = ((jde - pl->beg - (double)blk * pl->inc) + rel) / pl->inc;
 
         // the magick of function pointers

--- a/src/spk.c
+++ b/src/spk.c
@@ -247,7 +247,7 @@ int spk_calc(struct spk_s *pl, int m, double jde, double rel, struct mpos_s *pos
 	// find location of 'directory' describing the data records
 	n = (int)((jde + rel - pl->beg[m]) / pl->res[m]);
 	//val = (double*)pl->map + sizeof(double) * (pl->two[m][n] - 1);
-	val = (void *)pl->map + sizeof(double) * (pl->two[m][n] - 1);	
+	val = (double *)pl->map + pl->two[m][n] - 1;	
 
 	// record size and number of coefficients per coordinate
 	R = (int)val[-1];
@@ -257,8 +257,7 @@ int spk_calc(struct spk_s *pl, int m, double jde, double rel, struct mpos_s *pos
 	b = (int)(((jde - _jul(val[-3])) + rel) / (val[-2] / 86400.0));
 	//val = (double*)pl->map + sizeof(double) * (pl->one[m][n] - 1)
 	//+ sizeof(double) * b * R;
-	val = (void *)pl->map + sizeof(double) * (pl->one[m][n] - 1)
-			+ sizeof(double) * b * R;
+	val = (double *)pl->map + (pl->one[m][n] - 1) + b * R;
 
 	// scale to interpolation units
 	z = ((jde - _jul(val[0])) + rel) / (val[1] / 86400.0);

